### DISCUSSION TITLE
cuda response files (rsp) containing include paths

### DISCUSSION THREAD

**Discussion by soerenPeters** [clangd/clangd#1676] [Q&A]
**Situation**
 In my cmake file I enable the export compile commands in order to support `clangd` to resolve all the include-paths: 
`set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`
This works like a charm in plain c++, however for cuda files, cmake generates a so called "response file" (e.g. CUDA_includes.rsp). This response file contains all the necessary include paths for the compilation of the cuda files and is passed via `--options-file` to nvcc. 
The compilation of the cuda file works, however, clangd is not able to resolve the includes anymore.

**Workaround**
 I found the workaround that I can manually disable the generation of response files for the cuda language in cmake:
`set(CMAKE_CUDA_USE_RESPONSE_FILE_FOR_INCLUDES 0)`
If I do this, the includes are directly passed to nvcc and not via the --options-flag. Therefore clangd can resolve the includes.

I am wondering if those response files could be directly supported in clangd, or if it's a bug, or if my workaround is the way to go. looking forward to your answers. :)

**Comment by HighCommander4** [comment]
My sense is your workaround is the way to go. I don't think it would scale for clangd to have special support for options specific to compiler drivers like nvcc.

If you think it would make sense for clang itself to support `--options-file` for "flag compatibility" with nvcc, you're welcome to suggest that in an upstream issue at https://github.com/llvm/llvm-project/issues, and if that is added then clangd would get that support automatically.

Otherwise, the general idea for working with clang-based tools is to arrange for your build system to generate a `compile_commands.json` that contains all the flags that clang would need to understand your code.
